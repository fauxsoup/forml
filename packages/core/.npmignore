.*
src/
*.config.js
.log
test/
docs/
coverage/
stats.json
