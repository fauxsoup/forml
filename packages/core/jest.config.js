module.exports = async () => {
    return {
        rootDir: __dirname,
        reporters: ['default', 'jest-summary-reporter'],
        testPathIgnorePatterns: [
            '/node_modules/',
            '/lib/',
            '/decorator-bootstrap/',
        ],
        transformIgnorePatterns: ['/node_modules/(?!(@mui|@babel)/)'],
        testEnvironment: 'jsdom',
        collectCoverageFrom: ['src/**/*.js', 'src/**/*.jsx'],
        coveragePathIgnorePatterns: ['/__tests__/', '/node_modules/', '/lib/'],
    };
};
