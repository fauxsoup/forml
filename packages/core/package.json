{
  "name": "@forml/core",
  "version": "1.5.4",
  "description": "A JSON Schema-based form generator powered by React",
  "main": "lib/index.js",
  "scripts": {
    "lint": "eslint src/**/*.jsx",
    "build": "babel src --out-dir lib",
    "watch": "babel src --out-dir lib --watch",
    "prepublish": "npm run build",
    "test": "jest --coverage --verbose"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/forml-js/forml.git"
  },
  "keywords": [
    "react",
    "json",
    "schema",
    "jsonschema",
    "form",
    "form-generator",
    "generator"
  ],
  "author": "Zachary Hueras",
  "license": "MIT",
  "bugs": {
    "url": "https://github.com/forml-js/forml/issues"
  },
  "homepage": "https://www.forml.dev",
  "peerDependencies": {
    "react": "^16.14.0",
    "react-dom": "^16.14.0"
  },
  "dependencies": {
    "@forml/context": "^1.0.1",
    "@forml/hooks": "^1.0.3"
  },
  "devDependencies": {
    "@babel/cli": "^7.12.8",
    "@babel/core": "^7.12.8",
    "@babel/plugin-proposal-object-rest-spread": "^7.12.1",
    "@babel/preset-env": "^7.12.7",
    "@babel/preset-react": "^7.12.7",
    "ajv": "^6.12.6",
    "deepmerge": "^4.2.2",
    "objectpath": "^2.0.0",
    "prop-types": "^15.7.2",
    "react-beautiful-dnd": "^13.0.0",
    "shortid": "^2.2.16"
  },
  "gitHead": "5eac06f4aef45bd7c4ec09e965eb791e2cf7015b"
}
