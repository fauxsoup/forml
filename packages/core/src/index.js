/**
 * @namespace forml
 */

import * as util from "./util";
import * as constants from "./constants";

export * from "./schema-form";
export * from "./schema-field";

export * from "@forml/ware";

export { constants, util };
