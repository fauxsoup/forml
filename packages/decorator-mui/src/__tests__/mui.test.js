import { AdapterMoment } from '@mui/x-date-pickers/AdapterMoment';
import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
import { render } from '@testing-library/react';
import { SchemaForm, util } from '@forml/core';
import React from 'react';
import * as mui from '../';
import ThemeProvider from '@mui/material/styles/ThemeProvider';
import createTheme from '@mui/material/styles/createTheme';

const theme = createTheme({});

describe('Material UI', function () {
    describe('with no model', function () {
        const tests = [
            ['root string', { type: 'string' }],
            ['root number', { type: 'number' }],
            ['root integer', { type: 'integer' }],
            ['root date', { type: 'string', format: 'date' }],
            ['root date-time', { type: 'string', format: 'date-time' }],
            ['root enumeration', { type: 'string', enum: ['a', 'b', 'c'] }],
            [
                'root multiselect',
                {
                    type: 'array',
                    items: { type: 'string', enum: ['a', 'b', 'c'] },
                    uniqueItems: true,
                },
            ],
            //[
            //    'root tuple',
            //    { type: 'array', items: [{ type: 'string' }, { type: 'number' }] },
            //],
            //['root array', { type: 'array', items: { type: 'string' } }],
            [
                'root object',
                {
                    type: 'object',
                    properties: {
                        test: { type: 'string' },
                        nested: {
                            type: 'object',
                            properties: {
                                property: { type: 'number' },
                            },
                        },
                    },
                },
            ],
        ];

        const form = ['*'];
        const decorator = mui;

        for (let [name, schema] of tests) {
            test(`renders empty ${name} consistently`, function () {
                const model = util.defaultForSchema(schema);
                const { container } = render(
                    <ThemeProvider theme={theme}>
                        <LocalizationProvider dateAdapter={AdapterMoment}>
                            <SchemaForm
                                schema={schema}
                                form={form}
                                model={model}
                                decorator={decorator}
                            />
                        </LocalizationProvider>
                    </ThemeProvider>
                );

                expect(container).toMatchSnapshot();
            });
        }
    });
});
