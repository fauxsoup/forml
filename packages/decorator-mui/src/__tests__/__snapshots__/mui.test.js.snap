// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Material UI with no model renders empty root date consistently 1`] = `
<div>
  <div
    class="MuiFormControl-root css-1lq4zch-MuiFormControl-root"
  >
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary Mui-error MuiInputBase-formControl MuiInputBase-adornedEnd css-1ptx2yq-MuiInputBase-root-MuiInput-root"
    >
      <input
        aria-invalid="true"
        class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd css-1x51dt5-MuiInputBase-input-MuiInput-input"
        placeholder="mm/dd/yyyy"
        type="tel"
        value=""
      />
      <div
        class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-standard MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
      >
        <button
          aria-label="Choose date"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CalendarIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <p
      class="MuiFormHelperText-root Mui-error MuiFormHelperText-sizeMedium css-1d1r5q-MuiFormHelperText-root"
    >
      invalid schema
    </p>
  </div>
</div>
`;

exports[`Material UI with no model renders empty root date-time consistently 1`] = `
<div>
  <div
    class="MuiFormControl-root css-1lq4zch-MuiFormControl-root"
  >
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary Mui-error MuiInputBase-formControl MuiInputBase-adornedEnd css-1ptx2yq-MuiInputBase-root-MuiInput-root"
    >
      <input
        aria-invalid="true"
        class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd css-1x51dt5-MuiInputBase-input-MuiInput-input"
        placeholder="mm/dd/yyyy hh:mm (a|p)m"
        type="tel"
        value=""
      />
      <div
        class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-standard MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
      >
        <button
          aria-label="Choose date"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CalendarIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <p
      class="MuiFormHelperText-root Mui-error MuiFormHelperText-sizeMedium css-1d1r5q-MuiFormHelperText-root"
    >
      invalid schema
    </p>
  </div>
</div>
`;

exports[`Material UI with no model renders empty root enumeration consistently 1`] = `
<div>
  <div
    class="MuiFormControl-root css-1lq4zch-MuiFormControl-root"
  >
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary Mui-error MuiInputBase-formControl  css-a3l6o-MuiInputBase-root-MuiInput-root-MuiSelect-root"
    >
      <div
        aria-expanded="false"
        aria-haspopup="listbox"
        class="MuiSelect-select MuiSelect-standard MuiInputBase-input MuiInput-input css-1rxz5jq-MuiSelect-select-MuiInputBase-input-MuiInput-input"
        role="button"
        tabindex="0"
      >
        <span
          class="notranslate"
        >
          ​
        </span>
      </div>
      <input
        aria-hidden="true"
        class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
        tabindex="-1"
        value=""
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
        data-testid="ArrowDropDownIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </div>
    <p
      class="MuiFormHelperText-root Mui-error MuiFormHelperText-sizeMedium css-1d1r5q-MuiFormHelperText-root"
    >
      must be equal to one of the allowed values
    </p>
  </div>
</div>
`;

exports[`Material UI with no model renders empty root integer consistently 1`] = `
<div>
  <div
    class="MuiFormControl-root css-1lq4zch-MuiFormControl-root"
  >
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl css-vp6a25-MuiInputBase-root-MuiInput-root"
      form="[object Object]"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
        id=""
        type="text"
        value="0"
      />
    </div>
  </div>
</div>
`;

exports[`Material UI with no model renders empty root multiselect consistently 1`] = `
<div>
  <div
    class="MuiFormControl-root css-1lq4zch-MuiFormControl-root"
  >
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl  css-a3l6o-MuiInputBase-root-MuiInput-root-MuiSelect-root"
    >
      <div
        aria-expanded="false"
        aria-haspopup="listbox"
        class="MuiSelect-select MuiSelect-standard MuiSelect-multiple MuiInputBase-input MuiInput-input css-1rxz5jq-MuiSelect-select-MuiInputBase-input-MuiInput-input"
        role="button"
        tabindex="0"
      >
        <span
          class="notranslate"
        >
          ​
        </span>
      </div>
      <input
        aria-hidden="true"
        class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
        tabindex="-1"
        value=""
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
        data-testid="ArrowDropDownIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`Material UI with no model renders empty root number consistently 1`] = `
<div>
  <div
    class="MuiFormControl-root css-1lq4zch-MuiFormControl-root"
  >
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl css-vp6a25-MuiInputBase-root-MuiInput-root"
      form="[object Object]"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
        id=""
        type="text"
        value="0"
      />
    </div>
  </div>
</div>
`;

exports[`Material UI with no model renders empty root object consistently 1`] = `
<div>
  <ul
    class="MuiList-root MuiList-dense css-183iplg-MuiList-root"
  >
    <div
      class="MuiBox-root css-rwzwky"
    >
      <div
        class="MuiFormControl-root css-1lq4zch-MuiFormControl-root"
      >
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl css-vp6a25-MuiInputBase-root-MuiInput-root"
          form="[object Object]"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
            id="test"
            type="text"
            value=""
          />
        </div>
      </div>
      <ul
        class="MuiList-root MuiList-dense css-183iplg-MuiList-root"
      >
        <div
          class="MuiBox-root css-rwzwky"
        >
          <div
            class="MuiFormControl-root css-1lq4zch-MuiFormControl-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-colorPrimary MuiFormLabel-filled css-1c2i806-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
            >
              property
            </label>
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl css-vp6a25-MuiInputBase-root-MuiInput-root"
              form="[object Object]"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
                id="nested.property"
                type="text"
                value="0"
              />
            </div>
          </div>
        </div>
      </ul>
    </div>
  </ul>
</div>
`;

exports[`Material UI with no model renders empty root string consistently 1`] = `
<div>
  <div
    class="MuiFormControl-root css-1lq4zch-MuiFormControl-root"
  >
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl css-vp6a25-MuiInputBase-root-MuiInput-root"
      form="[object Object]"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
        id=""
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;
